src/progly/write-papertests.cc

```diff
@@ -0,0 +1,149 @@
+/*
+* Copyright (C) 2018 The Regents of the University of California
+* All Rights Reserved
+*
+* This library can redistribute it and/or modify under the terms
+* of the GNU Lesser General Public License Version 2.1 as published
+* by the Free Software Foundation.
+*
+*/
+
+/*
+bin/write-papertests ../src/progly/katfiles/dataset_arity1_1000000_rows-0.txt paper_exps 0 concat-data
+bin/write-papertests ../src/progly/katfiles/dataset_arity1_1000000_rows-0.txt paper_exps 1 separate-data
+*/
+
+#include <iostream>
+#include <ctime>
+#include <chrono>
+#include <fstream>
+#include <boost/program_options.hpp>
+
+#include "include/rados/librados.hpp"
+#include "include/cephfs/libcephfs.h"
+
+#include <string>
+#include <sstream>
+
+#include "include/types.h"
+#include <errno.h>
+#include <time.h>
+
+#include "objclass/objclass.h"
+
+namespace po = boost::program_options ;
+
+void raw_concat_write( std::string, std::string, std::string, bool ) ;
+void raw_separate_write( std::string, std::string, std::string, bool ) ;
+
+int main( int argc, char **argv ) {
+  std::cout << "in write-papertests..." << std::endl ;
+
+  // datetime code from https://stackoverflow.com/a/16358264
+  time_t rawtime ;
+  struct tm * timeinfo ;
+  char buffer[80] ;
+  time ( &rawtime ) ;
+  timeinfo = localtime( &rawtime ) ;
+  strftime( buffer, sizeof( buffer ),"%Y%m%d_%H%M%S", timeinfo ) ;
+  std::string str( buffer ) ;
+
+  std::string DATA_PATH    = argv[1] ;
+  std::string poolname     = argv[2] ;
+  int write_type           = std::atoi( argv[3] ) ;
+  std::string oidname      = argv[4] ;
+  bool debug               = true ;
+
+  std::ofstream res_file ;
+
+  if( debug ) {
+    std::cout << "================================" << std::endl ;
+    std::cout << "running with params:" << std::endl ;
+    std::cout << "DATA_PATH    = " << DATA_PATH << std::endl ;
+    std::cout << "poolname     = " << poolname << std::endl ;
+    std::cout << "write_type   = " << write_type << std::endl ;
+    std::cout << "oidname      = " << oidname << std::endl ;
+    std::cout << "timestamp    = " << str << std::endl ;
+    std::cout << "--->exp : " << oidname << std::endl ;
+  }
+
+  if( write_type == 0 ) {
+    raw_concat_write( oidname, DATA_PATH, poolname, debug ) ;
```

Comment:
Which are these - concat v. separate - is this for replace vs. append to obj?

Reply:
Nah. concat represents the storage strategy where all the integers are concatenated into a single string and then stored in the bl. separated represents the storage strategy where each integer is separately encoded into the bl.

Reply:
Ok I see.  These are for the binary raw data comparison tests.